geopandas/tests/test_geodataframe.py

```diff
@@ -62,13 +63,8 @@ def test_different_geo_colname(self):
         assert df._geometry_column_name == "location"
 
         geom2 = [Point(x, y) for x, y in zip(range(5, 10), range(5))]
-        df2 = df.set_geometry(geom2, crs="dummy_crs")
-        assert "location" in df2
-        assert df2.crs == "dummy_crs"
-        assert df2.geometry.crs == "dummy_crs"
-        # reset so it outputs okay
-        df2.crs = df.crs
-        assert_geoseries_equal(df2.geometry, GeoSeries(geom2, crs=df2.crs))
+        with pytest.raises(CRSError):
+            df2 = df.set_geometry(geom2, crs="dummy_crs")
```

Comment:
flake8 doesn't like this line as fails CI: `F841 local variable 'df2' is assigned to but never used`. 
```suggestion
            df.set_geometry(geom2, crs="dummy_crs")
```
This should be enough, no?